data/helpers.d/utils

```diff
@@ -29,55 +29,81 @@ ynh_get_plain_key() {
 #
 # The file conf/app.src need to contains:
 # 
-# SOURCE_URL=Address to download the app archive
-# SOURCE_SUM=Control sum
-# ARCH_FORMAT=tar.gz or zip
+# src_url="Address to download the app archive"
+# src_sum="Control sum"
+# src_format="tar.gz" # (Optional) default value: tar.gz
+# src_in_subdir=false # (Optional) Put false if source are directly in the archive root
+# src_filename="example.tar.gz" (Optionnal) default value: ${src_id}.${src_format}
 #
-# usage: ynh_setup_source [source_id]
+#
+# usage: ynh_setup_source dest_dir [source_id]
+# | arg: dest_dir  - Directory where to setup sources
 # | arg: source_id - Name of the app, if the package contains more than one app
 ynh_setup_source () {
-	source_id=${1:-app} # If the argument is not given, source_id equal "app"
-	src_url=$(cat ../conf/${source_id}.src | grep SOURCE_URL | cut -d= -f2-)
-	src_checksum=$(cat ../conf/${source_id}.src | grep SOURCE_SUM | cut -d= -f2-)
-	arch_format=$(cat ../conf/${source_id}.src | grep ARCH_FORMAT | cut -d= -f2-)
-	local_source="/opt/yunohost-apps-src/$YNH_APP_ID/source.$arch_format"
+	local dest_dir=$1
+    local src_id=${2:-app} # If the argument is not given, source_id equal "app"
+	# Default value
+	local src_format="tar.gz"
+    local src_sumprg="sha256sum"
+    local src_filename="${src_id}.${src_format}"
+    local src_in_subdir=true #false if source are directly in the archive root   
+
+    # Load value from configuration file (see above for a small doc about this file
+    # format)
+    source ../conf/${src_id}.src
+    src_format=$(echo "$src_format" | tr '[:upper:]' '[:lower:]')
+    if [ src_filename = "" ] ; then
+        src_filename="${src_id}.${src_format}"
+    fi
+    local local_src="/opt/yunohost-apps-src/${YNH_APP_ID}/${src_filename}"
 
 	if test -e "$local_source"
 	then	# Use the local source file if it is present
-		cp $local_source source.$arch_format
+		cp $local_src $src_filename
 	else	# If not, download the source
-		wget -nv -O source.$arch_format $src_url
+		wget -nv -O $src_filename $src_url
     fi
 
 	# Check the control sum
-	echo "$src_checksum source.$arch_format" \
-		| md5sum -c --status || ynh_die "Corrupt source"
+	echo "$src_checksum $src_filename" \
+		| ${src_sumprg} -c --status || ynh_die "Corrupt source"
```

Comment:
Here again, it's really not clear than $src_sumprg can be overwritten by the source file !